src/Sim.tsx

```diff
@@ -111,116 +109,82 @@ export class Space {
     });
   };
 
+  private lens_: Babylon.LensRenderingPipeline;
+
   
   private lastState_: State = undefined;
   private debounceUpdate_ = false;
+
+  private sceneSetting_: boolean;
+  private latestUnfulfilledScene_: Scene;
+
   private onStoreChange_ = (state: State) => {
     if (this.debounceUpdate_) {
       this.lastState_ = state;
```

Comment:
### Object "reset" is broken

Should be easy to repro with the editable box in the current test scene. Clicking "reset" once, then click it more times.

This is because changes in the Babylon scene aren't applied to the `SceneBinding`'s understanding of the scene. After the first reset, gravity makes the box fall. Since the box's position is changing, `Sim` sends updates to Redux, but `SceneBinding` doesn't know about these, so it thinks the box is still in the air. So clicking "reset" again has no effect since the patch is empty.

Might be as easy as updating `SceneBinding`'s scene here (instead of `this.lastState_`, which seems unused anyway). But I'm not sure if that's the "correct" fix

Reply:
Fixed via PR into this branch